addons/udes_sale_stock/models/sale_order.py

```diff
@@ -176,6 +178,30 @@ def check_state_cancelled(self):
                 to_cancel |= order
         to_cancel.write({'state': 'cancel'})
 
+    @api.model
+    def confirm_if_due(self):
+        """
+        Confirm sale orders in self that are due to be confirmed
+        If no orders passed into self, will confirm all unconfirmed orders
+
+        Returns recordset of orders where confirmation was attempted
+        """
+        days = self.env.ref('stock.warehouse0').u_so_auto_confirm_ahead_days
+        unconfirmed_states = ('draft', 'sent')
+        unconfirmed_so = self or self.search(
+            [('state', 'in', unconfirmed_states)]
+        )
+
+        # If ahead days set to -1, confirm all.
+        if days == -1:
+            return self.action_confirm()
```

Comment:
Is this supposed to be `unconfirmed_so.action_confirm()` ?

Reply:
yes! thank you